Mathlib/CategoryTheory/Limits/Preserves/Shapes/Pullbacks.lean

```diff
@@ -0,0 +1,328 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta, Andrew Yang
+
+! This file was ported from Lean 3 source module category_theory.limits.preserves.shapes.pullbacks
+! leanprover-community/mathlib commit f11e306adb9f2a393539d2bb4293bf1b42caa7ac
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.CategoryTheory.Limits.Shapes.Pullbacks
+import Mathlib.CategoryTheory.Limits.Preserves.Basic
+
+/-!
+# Preserving pullbacks
+
+Constructions to relate the notions of preserving pullbacks and reflecting pullbacks to concrete
+pullback cones.
+
+In particular, we show that `pullbackComparison G f g` is an isomorphism iff `G` preserves
+the pullback of `f` and `g`.
+
+The dual is also given.
+
+## TODO
+
+* Generalise to wide pullbacks
+
+-/
+
+
+noncomputable section
+
+universe v₁ v₂ u₁ u₂
+
+-- Porting note: need Functor namespace for mapCone
+open CategoryTheory CategoryTheory.Category CategoryTheory.Limits CategoryTheory.Functor
+
+namespace CategoryTheory.Limits
+
+section Pullback
+
+variable {C : Type u₁} [Category.{v₁} C]
+
+variable {D : Type u₂} [Category.{v₂} D]
+
+variable (G : C ⥤ D)
+
+variable {W X Y Z : C} {f : X ⟶ Z} {g : Y ⟶ Z} {h : W ⟶ X} {k : W ⟶ Y} (comm : h ≫ f = k ≫ g)
+
+/-- The map of a pullback cone is a limit iff the fork consisting of the mapped morphisms is a
+limit. This essentially lets us commute `PullbackCone.mk` with `Functor.mapCone`. -/
+def isLimitMapConePullbackConeEquiv :
+    IsLimit (mapCone G (PullbackCone.mk h k comm)) ≃
+      IsLimit
+        (PullbackCone.mk (G.map h) (G.map k) (by simp only [← G.map_comp, comm]) :
+          PullbackCone (G.map f) (G.map g)) :=
+  (IsLimit.postcomposeHomEquiv (diagramIsoCospan.{v₂} _) _).symm.trans <|
+    IsLimit.equivIsoLimit <|
+      Cones.ext (Iso.refl _) <| by
+        rintro (_ | _ | _) <;> dsimp <;> simp only [comp_id, id_comp, G.map_comp]
+#align category_theory.limits.is_limit_map_cone_pullback_cone_equiv CategoryTheory.Limits.isLimitMapConePullbackConeEquiv
+
+/-- The property of preserving pullbacks expressed in terms of binary fans. -/
+def isLimitPullbackConeMapOfIsLimit [PreservesLimit (cospan f g) G]
+    (l : IsLimit (PullbackCone.mk h k comm)) : 
+    -- Porting note: help figure out the hole
```

Comment:
Please make this note a bit more explicit. I don't understand it.

Reply:
Oh. This one probably doesn't rise to the level of a porting note. I had to fill in `h` and `k` similarly for the others.

Reply:
Ok, in that case, please remove the note (-;

Reply:
Removed